qxf2/QuarksB
# Test to verify staging-newsletter-SQS received message

How to run the test:
1. Checkout to the Branch-
2. In python virtual environment 3, install requirements.
3. Run the Test using command-
4. Note before running test, you will have to setup ~/.aws/credentials- This test ran successfully for `lambda-tester` user.

` python -m pytest tests/test_send_and_validate_message_from_queue.py -sv`

4. You will see output as below:
Output:

```
rahulbhave@DESKTOP-VCILPB0 MINGW64 ~/code/QuarksB (rahul-verify-on-sqs)
$ python -m pytest tests/test_send_and_validate_message_from_queue.py -sv
============================= test session starts =============================
platform win32 -- Python 3.9.6, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- C:\Users\rahulbhave\code\QuarksB\venv3\Scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\rahulbhave\code\QuarksB
collecting ... collected 1 item

tests/test_send_and_validate_message_from_queue.py::TestSQSMessage::test_send_and_validate_message_from_queue {"message": "Setup- Purge SQS queue"}
{"message": "---------------------------------------------------------------------------"}
{"message": "Found credentials in shared credentials file: ~/.aws/credentials"}
{"message": "Sending the message"}
{"message": "---------------------------------------------------------------------------"}
{"message": "Getting message from SQS Queue"}
{"message": "---------------------------------------------------------------------------"}
{"message": "Validating message from SQS Queue"}
{"message": "---------------------------------------------------------------------------"}
PASSED

============================== 1 passed in 4.78s ==============================

```
